components/feature/prompts/src/main/java/mozilla/components/feature/prompts/PromptFeature.kt

```diff
@@ -758,20 +777,80 @@ class PromptFeature private constructor(
     }
 }
 
+/**
+ * Removes the [PromptRequest] indicated by [promptRequestUID] from the current Session if it it exists
+ * and offers a [consume] callback for other optional side effects.
+ *
+ * @param sessionId Session id of the tab or custom tab in which to try consuming [PromptRequests].
+ * If the id is not provided or a tab with that id is not found the method will act on the current tab.
+ * @param promptRequestUID Id of the [PromptRequest] to be consumed.
+ * @param activePrompt The current active Prompt if known. If provided it will always be cleared,
+ * irrespective of if [PromptRequest] indicated by [promptRequestUID] is found and removed or not.
+ * @param consume callback with the [PromptRequest] if found, before being removed from the Session.
+ */
 internal fun BrowserStore.consumePromptFrom(
     sessionId: String?,
+    promptRequestUID: String,
     activePrompt: WeakReference<PromptDialogFragment>? = null,
     consume: (PromptRequest) -> Unit
 ) {
-    if (sessionId == null) {
```

Comment:
This if/else check is already done in the `findTabOrCustomTabOrSelectedTab` method.